applications/GeoMechanicsApplication/tests/cpp_tests/custom_utilities/test_stress_strain_utitlities.cpp

```diff
@@ -190,4 +192,62 @@ KRATOS_TEST_CASE_IN_SUITE(CheckCalculateStrains, KratosGeoMechanicsFastSuiteWith
         KRATOS_EXPECT_VECTOR_NEAR(strains[i], expected_strains[i], 1.E-6)
 }
 
+KRATOS_TEST_CASE_IN_SUITE(CheckCalculatePrincipalStresses, KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    Vector cauchy_stresses = ZeroVector(6);
+    cauchy_stresses <<= 80.0, 50.0, 20.0, 40.0, 35.0, 45.0;
+
+    Vector principal_stresses;
+    Matrix eigenvectors_matrix;
+    StressStrainUtilities::CalculatePrincipalStresses(cauchy_stresses, principal_stresses, eigenvectors_matrix);
+
+    Vector expected_solution = ZeroVector(3);
+    expected_solution <<= 135.736961146391, 22.5224297324582, -8.25939087884923;
+
+    constexpr double tolerance{1.0e-12};
+
+    for (unsigned int i = 0; i < principal_stresses.size(); ++i) {
+        KRATOS_EXPECT_NEAR(principal_stresses(i), expected_solution(i), tolerance);
+    }
```

Comment:
I think we can simplify this a bit:
```suggestion
    Vector expected_solution(3);
    expected_solution <<= 135.736961146391, 22.5224297324582, -8.25939087884923;
    KRATOS_EXPECT_VECTOR_NEAR(principal_stresses, expected_solution, Defaults::absolute_tolerance)
```